client/cmd/list.go

```diff
@@ -0,0 +1,291 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hoophq/hoop/client/cmd/styles"
+	clientconfig "github.com/hoophq/hoop/client/config"
+	"github.com/hoophq/hoop/common/httpclient"
+	"github.com/hoophq/hoop/common/log"
+	"github.com/hoophq/hoop/common/version"
+	"github.com/spf13/cobra"
+)
+
+var (
+	verboseFlag bool
+	outputFlag  string
+)
+
+var listExampleDesc = `hoop list
+hoop ls
+hoop list --verbose
+hoop list -o json`
+
+var listCmd = &cobra.Command{
+	Use:     "list",
+	Aliases: []string{"ls"},
+	Short:   "List available connections",
+	Long:    "Display connections that you have access to connect",
+	Example: listExampleDesc,
+	Run: func(cmd *cobra.Command, args []string) {
+		runList()
+	},
+}
+
+func init() {
+	listCmd.Flags().BoolVarP(&verboseFlag, "verbose", "v", false, "Show additional information including command")
+	listCmd.Flags().StringVarP(&outputFlag, "output", "o", "", "Output format. One of: (json)")
+	rootCmd.AddCommand(listCmd)
+}
+
+func runList() {
+	config := clientconfig.GetClientConfigOrDie()
+
+	connections, err := fetchConnections(config)
+	if err != nil {
+		styles.PrintErrorAndExit("Failed to fetch connections: %v", err)
+	}
+
+	if outputFlag == "json" {
+		jsonData, err := json.MarshalIndent(connections, "", "  ")
+		if err != nil {
+			styles.PrintErrorAndExit("Failed to encode JSON: %v", err)
+		}
+		fmt.Print(string(jsonData))
+		return
+	}
+
+	displayConnections(connections)
+}
+
+func fetchConnections(config *clientconfig.Config) ([]map[string]any, error) {
+	url := config.ApiURL + "/api/connections"
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating request: %v", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %v", config.Token))
+	if config.IsApiKey() {
+		req.Header.Set("Api-Key", config.Token)
+	}
+	req.Header.Set("User-Agent", fmt.Sprintf("hoopcli/%v", version.Get().Version))
+
+	resp, err := httpclient.NewHttpClient(config.TlsCA()).Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed performing request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	log.Debugf("http response %v", resp.StatusCode)
+
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("failed to fetch connections, status=%v", resp.StatusCode)
+	}
+
+	var connections []map[string]any
+	if err := json.NewDecoder(resp.Body).Decode(&connections); err != nil {
+		return nil, fmt.Errorf("failed decoding response: %v", err)
+	}
```

Comment:
Make sure to return a generic error for calls to the api, this will make sure that we all catch any non success error along with the body, example:

```sh
	resp, err := httpclient.NewHttpClient(config.TlsCA()).Do(req)
	if err != nil {
		return nil, nil, err
	}
	log.Debugf("http response %v", resp.StatusCode)
	defer resp.Body.Close()
	if resp.StatusCode != 200 {
		respBody, _ := io.ReadAll(resp.Body)
		return nil, resp.Header, fmt.Errorf("failed performing request, status=%v, body=%v",
			resp.StatusCode, string(respBody))
	}
```